src/common/transformations/src/transformations/common_optimizations/concat_to_broadcast.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/concat_to_broadcast.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::ConcatToBroadcast::ConcatToBroadcast() {
+    MATCHER_SCOPE(ConcatToBroadcast);
+
+    auto concat_label = pattern::wrap_type<op::v0::Concat>([](const Output<Node>& value) {
+        auto node = value.get_node_shared_ptr();
+        auto first_input_source_output = node->get_input_source_output(0);
+
+        // Do not start comparing from 0 (with itself)
+        const auto& input_values = node->input_values();
+        auto it = input_values.cbegin();
+        std::next(it, 1);
+
+        for (; it != input_values.cend(); ++it) {
+            if (*it != first_input_source_output) {
+                return false;
+            }
+        }
+
+        return true;
+    });
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = pattern_map.at(concat_label).get_node_shared_ptr();
+
+        if (transformation_callback(concat)) {
+            return false;
+        }
+
+        if (concat->get_default_output().get_partial_shape().is_dynamic()) {
+            return false;
+        }
+
+        const auto& broadcast_input = concat->input_value(0);
+        auto target_shape =
+            std::make_shared<ov::op::v0::Constant>(ov::element::i32,
+                                                   Shape{concat->get_default_output().get_shape().size()},
+                                                   concat->get_default_output().get_shape());
+        auto broadcast = std::make_shared<ov::op::v3::Broadcast>(broadcast_input, target_shape);
```

Comment:
in general case it have to be Tile operation,
but or case when concatenation dim is 1, it can be Broadcast

It looks like we need to check, the original dim, if it's 1 when insert Broadcast, otherwise we need to insert Tile

e.g.
in_1 shape = 1, 2, 3
in_2 shape = 1, 2, 3
etc 

if concat axis is 0, it corresponds to in_N_shape[0] -> 1
but if the axis is equal to 1, when in_N_shape[1] -> 2 , we can't use Broadcast in this case

Reply:
Added Tile.